C7/Map/TntLayer.cs

```diff
@@ -0,0 +1,42 @@
+using C7GameData;
+using Godot;
+using Resource = C7GameData.Resource;
+using Serilog;
+
+namespace C7.Map
+{
+	/// <summary>
+	/// Displays terrain yield overlays (from the tnt.pcx file).  These are most well known for letting you know where
+	/// there are bonus grasslands.
+	/// Note: I don't know why it's called tnt.
+	/// </summary>
+	public class TntLayer : LooseLayer
+	{
+		private ILogger log = LogManager.ForContext<ResourceLayer>();
+
+		private static readonly Vector2 tntSize = new Vector2(128, 64);
+		private ImageTexture txtTexture;
+
+		//Each row corresponds to a terrain.  For now we're only adding one, maybe someday we'll add full TNT support
+		private readonly int GRASSLAND_ROW = 0;
+		private readonly int BONUS_GRASSLAND_ROW = 1;
+		private readonly int PLAINS_ROW = 2;
+		private readonly int DESERT_ROW = 3;
+		private readonly int BONUS_GRASSLAND_TNT_OFF_ROW = 3;
+		private readonly int TUNDRA_ROW = 4;
+		private readonly int FLOOD_PLAIN_ROW = 5;
+
+		public TntLayer()
+		{
+			txtTexture = Util.LoadTextureFromPCX("Art/Terrain/tnt.pcx");
+		}
+		public override void drawObject(LooseView looseView, GameData gameData, Tile tile, Vector2 tileCenter)
+		{
+			if (tile.overlayTerrainType.Key == "grassland" && tile.isBonusShield) {
+				Rect2 tntRectangle = new Rect2(0 * tntSize.x, BONUS_GRASSLAND_TNT_OFF_ROW * tntSize.y, tntSize);
```

Comment:
the first parameter will always be 0

Reply:
True.  In Civ3, it alternates between 0 and 2 depending on some formula based on the tile, probably similar to forests/mountains.  I've just left it as 0 for now in lieu of a formula, the differences are miniscule anyway.  It now is always 0, no multiplication needed.